From 8e4c150443a84cf929976a647e28c7ae3fec210e Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:47 -0400
Subject: [misc] mei: remove unnecessary pci_set_drvdata()

Message-id: <1399041470-21330-15-git-send-email-prarit@redhat.com>
Patchwork-id: 79301
O-Subject: [RHEL7.1 PATCH BZ 1088953 14/77] misc: mei: remove unnecessary pci_set_drvdata()
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit b9be2c60fce7a93723403488a2f5d508ff7d3d2f
Author: Jingoo Han <jg1.han@samsung.com>
Date:   Mon Sep 23 10:37:24 2013 +0900

    misc: mei: remove unnecessary pci_set_drvdata()

    The driver core clears the driver data to NULL after device_release
    or on probe failure. Thus, it is not needed to manually clear the
    device driver data to NULL.

    Signed-off-by: Jingoo Han <jg1.han@samsung.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/pci-me.c b/drivers/misc/mei/pci-me.c
index 3c9e257..ec64eab 100644
--- a/drivers/misc/mei/pci-me.c
+++ b/drivers/misc/mei/pci-me.c
@@ -277,7 +277,6 @@ static void mei_me_remove(struct pci_dev *pdev)
 
  free_irq(pdev->irq, dev);
  pci_disable_msi(pdev);
- pci_set_drvdata(pdev, NULL);
 
  if (hw->mem_addr)
   pci_iounmap(pdev, hw->mem_addr);
-- 
1.7.1